ExilenceNextApp/src/components/table-wrapper/TableWrapper.tsx

```diff
@@ -0,0 +1,137 @@
+import TableContainer from '@material-ui/core/TableContainer';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+import clsx from 'clsx';
+import React, { CSSProperties, useEffect } from 'react';
+import { Cell, HeaderGroup, Meta, Row, TableInstance } from 'react-table';
+import { useDebounce } from '../../hooks/use-debounce';
+import { ResizeHandle } from '../resize-handle/ResizeHandle';
+import { TablePagination } from '../table-pagination/TablePagination';
+import { useStyles } from './TableWrapper.styles';
+
+interface ITableWrapperProps {
+  instance: TableInstance<object>;
+  onClick?: (row: Row<object>) => void;
+  setInitialState: any;
+}
+
+const getStyles = <T extends object>(props: any, disableResizing = false, align = 'left') => [
+  props,
+  {
+    style: {
+      justifyContent: align === 'right' ? 'flex-end' : 'flex-start',
+      alignItems: 'flex-start',
+      display: 'flex',
+    },
+  },
+];
+
+const headerProps = <T extends object>(
+  props: any,
+  { column }: Meta<T, { column: HeaderGroup<T> }>
+) => getStyles(props, column && column.disableResizing, column && column.align);
+
+const cellProps = <T extends object>(props: any, { cell }: Meta<T, { cell: Cell<T> }>) =>
+  getStyles(props, cell.column && cell.column.disableResizing, cell.column && cell.column.align);
+
+const TableWrapper = ({ instance, onClick, setInitialState }: ITableWrapperProps) => {
+  const classes = useStyles();
+
+  const {
+    getTableProps,
+    headerGroups,
+    prepareRow,
+    page,
+    gotoPage,
+    setPageSize,
+    getTableBodyProps,
+    state,
+  } = instance;
+
+  const debouncedState = useDebounce(state, 500);
+
+  useEffect(() => {
+    const { sortBy, filters, pageSize, columnResizing, hiddenColumns } = debouncedState;
+    const val = {
+      sortBy,
+      filters,
+      pageSize,
+      columnResizing,
+      hiddenColumns,
+    };
+    setInitialState(val);
+  }, [setInitialState, debouncedState]);
+
+  const cellClickHandler = (cell: Cell<object>) => () => {
+    onClick && cell.column.id !== '_selector' && onClick(cell.row);
+  };
+
+  return (
+    <TableContainer className={classes.container}>
+      <div className={classes.tableTable} {...getTableProps()}>
+        <div>
+          {headerGroups.map((headerGroup, i) => (
+            <div {...headerGroup.getHeaderGroupProps()} className={classes.tableHeadRow} key={i}>
+              {headerGroup.headers.map((column, j) => {
+                const style = {
+                  textAlign: column.align ? column.align : 'left ',
+                } as CSSProperties;
+                return (
+                  <div
+                    {...column.getHeaderProps(headerProps)}
+                    className={classes.tableHeadCell}
+                    key={j}
```

Comment:
This may be a performance issue when react will update VDOM. We have the possibility of the same keys in above `key={i}` and here `key={j}`

Kindly reminder of 

![image](https://user-images.githubusercontent.com/19862545/96743920-c6842980-13c4-11eb-9f5f-b8441d4d3a32.png)

https://reactjs.org/docs/reconciliation.html#tradeoffs